sdk/core/core-amqp/src/util/cancellableDelay.ts

```diff
@@ -0,0 +1,17 @@
+import { AbortSignalLike, AbortError } from "@azure/abort-controller";
+
+export function cancellableDelay(delayInMs: number, abortSignal?: AbortSignalLike): Promise<void> {
+  return new Promise((resolve, reject) => {
+    if (abortSignal && abortSignal.aborted) {
+      return reject(new AbortError(`The delay was cancelled by the user.`));
```

Comment:
When a retry operation is cancelled during the wait time, this error message is not helpful to the user.
I would suggest to update the function to take a 3rd optional parameter where the caller can provide the error message to be used by the `AbortError`